webapp/graphite/remote_storage.py

```diff
@@ -139,12 +139,17 @@ def fetch(self, startTime, endTime, now=None, result_queue=None, headers=None):
       query_params.append(('now', str( int(now) )))
     query_string = urlencode(query_params)
 
-    connection = HTTPConnectionWithTimeout(self.store.host)
-    connection.timeout = settings.REMOTE_STORE_FETCH_TIMEOUT
-    if settings.REMOTE_STORE_USE_POST:
-      connection.request('POST', '/render/', query_string, headers)
-    else:
-      connection.request('GET', '/render/?' + query_string, None, headers)
+    try:
+      connection = HTTPConnectionWithTimeout(self.store.host)
+      connection.timeout = settings.REMOTE_STORE_FETCH_TIMEOUT
+      if settings.REMOTE_STORE_USE_POST:
+        connection.request('POST', '/render/', query_string, headers)
+      else:
+        connection.request('GET', '/render/?' + query_string, None, headers)
+    except:
+      self.store.fail()
+      return None
```

Comment:
I see there's a similar pattern above with the `fail()` and `return None`. Is there some additional information you determined in debugging this taht would be helpful context here? Is `None` the right thing to return? Is there a richer/more correct type?